sdk/eventhub/azure-eventhubs/azure/eventhub/_eventprocessor/partition_context.py

```diff
@@ -15,14 +15,31 @@ class PartitionContext(object):
     Users can use update_checkpoint() of this class to save checkpoint data.
     """
     def __init__(self, fully_qualified_namespace, eventhub_name, consumer_group_name,
-                 partition_id, owner_id, partition_manager=None):
+                 partition_id, partition_manager=None):
         # type: (str, str, str, str, str, PartitionManager) -> None
         self.fully_qualified_namespace = fully_qualified_namespace
         self.partition_id = partition_id
         self.eventhub_name = eventhub_name
         self.consumer_group_name = consumer_group_name
-        self.owner_id = owner_id
         self._partition_manager = partition_manager
+        self._last_received_event = None
+
+    @property
+    def last_enqueued_event_properties(self):
+        """
+        The latest enqueued event information. This property will be updated each time an event is received when
+        the receiver is created with `track_last_enqueued_event_properties` being `True`.
+        The dict includes following information of the partition:
+
+            - `sequence_number`
+            - `offset`
+            - `enqueued_time`
+            - `retrieval_time`
+
+        :rtype: dict or None
+        """
+        if self._last_received_event:
+            return self._last_received_event._get_last_enqueued_event_properties()  # pylint: disable=protected-access
```

Comment:
Rather than keeping this as a private function on EventData that we then have to pylint:disable, we could move it out into a separate util function.
```python
def get_last_enqueued_event_properties(event_data):
    """Extracts the last enqueued event in from the received event delivery annotations.
    :rtype: Dict[str, Any]
    """
    if event_data.message.delivery_annotations:
        sequence_number = event_data.message.delivery_annotations.get(PROP_LAST_ENQUEUED_SEQUENCE_NUMBER, None)
        enqueued_time_stamp = event_data.message.delivery_annotations.get(PROP_LAST_ENQUEUED_TIME_UTC, None)
        if enqueued_time_stamp:
            enqueued_time_stamp = utc_from_timestamp(float(enqueued_time_stamp)/1000)
        retrieval_time_stamp = event_data.message.delivery_annotations.get(PROP_RUNTIME_INFO_RETRIEVAL_TIME_UTC, None)
        if retrieval_time_stamp:
            retrieval_time_stamp = utc_from_timestamp(float(retrieval_time_stamp)/1000)
        offset_bytes = event_data.message.delivery_annotations.get(PROP_LAST_ENQUEUED_OFFSET, None)
        offset = offset_bytes.decode('UTF-8') if offset_bytes else None
        return {
            "sequence_number": sequence_number,
            "offset": offset,
            "enqueued_time": enqueued_time_stamp,
            "retrieval_time": retrieval_time_stamp
        }
    return None
```

Reply:
done, add `get_last_enqueued_event_properties` method in the _utils.py